### DISCUSSION TITLE
Image from data or UIImage

### DISCUSSION THREAD

**Discussion by Louis-PAGNIER** [skiptools/skip#125] [Q&A] [answered]
Hello !
Is there any way to create an Image from stored data or UIImage with Skip ?

**Comment by marcprux** [comment]
If your image is stored in an assert catalog or in a URL in your app's Bundle, you can load it through one of the methods described at https://skip.tools/docs/modules/skip-ui/#images

We don't currently implement `UIImage` in SkipUI. If this is an API you need/want, can you outline your use case? We might be able to add some limited support, or  suggest an alternative implementation.

**Reply by Louis-PAGNIER** [reply]
Thank you for your reply.

My images are not defined, instead the user select them from a view using Pixabay or a PhotosPicker. Currently they are rescale then stored as part of a struct with SwiftData. For the storage I plan to switch to SkipSql, but for images I don't know if it is possible to load them from bytes like in SwiftUI.

**Reply by marcprux** [reply, marked as answer]
There's unfortunately no `SwiftUI.Image(data:)` constructor, so if you are wanting to load an image from a Data or byte array, your best bet right now would be to store it to a local URL and then load it with `AsyncImage(url:)`. Something like:

```swift
let data: Data = picker.imageData
let url: URL = URL.temporaryDirectory.appendingPathComponent("my-image-\(UUID().uuidString).png")
try data.write(to: url)

// now display the image in a view
AsyncImage(url: url) 
```

If you don't want to write to the file system, you could also construct a "data:" URL with the base64-encoded bytes, which should also work with `AsyncImage`.

Let us know if either of these work for you.

**Reply by Louis-PAGNIER** [reply]
Thank you, I will try next week to port my app to Skip, I was currently just informing if everything would be possible.
I also have a last question, is PhotosPicker planned to be implemented with Skip ?
I tried in playground and it seems not recognized.

**Reply by marcprux** [reply]
> I also have a last question, is PhotosPicker planned to be implemented with Skip ?

As it happens, we just released a preview package called [SkipKit](https://skip.tools/docs/modules/skip-kit/#camera-and-media-selection) that provides very rudimentary supports for activating the local media picker for the device. Please give it a try and let us know if it works for you. We expect to be making enhancements to this package based on user feedback.
